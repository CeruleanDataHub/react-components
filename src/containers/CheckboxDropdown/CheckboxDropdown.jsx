import PropTypes from "prop-types";
import React, { useState } from "react";
import styled from "styled-components";

import { Checkbox } from "../Checkbox/Checkbox";
import { Dropdown } from "../Dropdown/Dropdown";

const DropdownItem = styled.li`
  padding-top: 1em;
  padding-left: 0.5em;
`;

const DropdownText = styled.span`
  color: ${({ theme }) => theme.green};
  font-size: 1.25em;
  font-weight: bold;
`;

const generateList = (items, handleSelection) =>
  items.map(({ value, label, checked }) => (
    <DropdownItem key={label}>
      <Checkbox
        checked={checked}
        value={value}
        label={label}
        onChange={handleSelection}
      />
    </DropdownItem>
  ));

const setSelectedItemAsChecked = value => items =>
  items.map(item => ({
    ...item,
    checked: item.value === value ? !item.checked : item.checked
  }));

const Label = ({ label, items, isOpen }) => (
  <>
    {label}
    {isOpen && <DropdownText>{`${items.length} selected`}</DropdownText>}
  </>
);

Label.propTypes = {
  /** Define dropdown visibility state */
  isOpen: Dropdown.propTypes.isOpen,
  /** Text to show in dropdown */
  label: Dropdown.propTypes.label,
  /** Handler function */
  /** Items to be shown in dropdown list */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string
    })
  )
};

Label.defaultProps = {
  isOpen: false,
  label: "",
  items: []
};

export const CheckboxDropdown = ({ items, isOpen, label, onClick }) => {
  const [selectedItems, setSelectedItems] = useState(
    items.map(item => ({ ...item, checked: false }))
  );

  const checkedItems = selectedItems.filter(({ checked }) => checked);

  const handleSelection = value => {
    setSelectedItems(setSelectedItemAsChecked(value));
  };

  const dropdownLabel = (
    <Label label={label} items={checkedItems} isOpen={isOpen} />
  );

  return (
    <Dropdown label={dropdownLabel} isOpen={isOpen} onClick={onClick}>
      {generateList(selectedItems, handleSelection)}
    </Dropdown>
  );
};

CheckboxDropdown.propTypes = {
  /** Define dropdown visibility state */
  isOpen: Dropdown.propTypes.isOpen,
  /** Text to show in dropdown */
  label: Dropdown.propTypes.label,
  /** Handler function */
  onClick: Dropdown.propTypes.onClick,
  /** Items to be shown in dropdown list */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string
    })
  )
};

CheckboxDropdown.defaultProps = {
  isOpen: Dropdown.defaultProps.isOpen,
  label: Dropdown.defaultProps.label,
  onClick: Dropdown.defaultProps.onClick,
  items: []
};
