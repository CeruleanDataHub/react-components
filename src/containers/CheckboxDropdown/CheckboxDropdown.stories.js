import { action } from "@storybook/addon-actions";
import { withKnobs } from "@storybook/addon-knobs";
import React, { useState } from "react";

import { CheckboxDropdown } from "./CheckboxDropdown";

export default {
  title: "CheckboxDropdown",
  component: CheckboxDropdown,
  decorators: [withKnobs]
};

const items = [
  { value: "1", label: "Item 1" },
  { value: "2", label: "Item 2" },
  { value: "3", label: "Item 3" },
  { value: "4", label: "Item 4" },
  { value: "5", label: "Item 5" }
];

export const CheckboxDropdownStory = () => (
  <div style={{ width: 300 }}>
    <CheckboxDropdown label="Label" />
  </div>
);

export const CheckboxDropdownOpenStory = () => (
  <div style={{ minHeight: 250, width: 300 }}>
    <CheckboxDropdown label="Label" isOpen items={items} />
  </div>
);

export const CheckboxDropdownToggleStory = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div style={{ minHeight: 250, width: 300 }}>
      <CheckboxDropdown
        label="Label"
        items={items}
        isOpen={isOpen}
        onClick={() => {
          setIsOpen(!isOpen);
          action("dropdown-open");
        }}
      />
    </div>
  );
};
