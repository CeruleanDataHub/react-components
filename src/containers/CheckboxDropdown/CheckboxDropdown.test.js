import { mount } from "enzyme";
import React from "react";

import { CheckboxDropdown } from "./CheckboxDropdown";

const items = [
  { value: "some value", label: "Some item" },
  { value: "some other value", label: "Some other item" }
];

describe("CheckboxDropdown", () => {
  let component;
  let onClickMock;

  beforeEach(() => {
    onClickMock = jest.fn();

    component = mount(
      <CheckboxDropdown
        label="hello"
        items={items}
        isOpen
        onClick={onClickMock}
      />
    );
  });

  it("renders", () => {
    expect(component.render()).toMatchSnapshot();
  });

  it("does not call onClick yet", () => {
    expect(onClickMock).not.toHaveBeenCalled();
  });

  it("when clicked, callback is called", () => {
    component.find("button").simulate("click");

    expect(onClickMock).toHaveBeenCalled();
  });

  it("should be able to select item from list", () => {
    component
      .find("input")
      .first()
      .simulate("change");

    expect(component.render()).toMatchSnapshot();
  });
});
