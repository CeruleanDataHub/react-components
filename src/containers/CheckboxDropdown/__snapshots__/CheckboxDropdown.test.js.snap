// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxDropdown should be able to select item from list 1`] = `
.c6 {
  text-align: center;
}

.c10 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c16 {
  display: inline-block;
  width: 1em;
  height: 1em;
  border: 1px solid #999;
  background: transparent;
  text-align: center;
  border-radius: 4px;
  line-height: 1em;
}

.c12 {
  display: inline-block;
  width: 1em;
  height: 1em;
  border: 1px solid;
  text-align: center;
  border-radius: 4px;
  line-height: 1em;
}

.c13 {
  font-size: 0.75em;
  font-weight: bold;
}

.c15 {
  color: #999;
  margin-left: 1.5em;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  margin: 0.25em 0;
}

.c9:hover .c11 {
  border-color: #333;
}

.c9:hover .c14 {
  color: #666;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c7 {
  width: 100%;
  padding: 0;
  border: 1px solid;
  border-top: none;
  position: absolute;
  top: 22px;
  border-radius: 0 0 4px 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
  z-index: 1;
  box-sizing: border-box;
}

.c5 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c8 {
  padding-top: 1em;
  padding-left: 0.5em;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      <span
        className="c4"
      >
        Array [
          "",
          .c0 {
  font-size: 1.25em;
  font-weight: bold;
}

<span
            className="c0"
          >
            1 selected
          </span>,
        ]
      </span>
      <span
        className="c5 c6 lni lni-chevron-up"
      />
    </button>
    <ul
      className="c7"
      data-item-list-test={true}
    >
      <li
        className="c8"
      >
        <label
          className="c9"
        >
          <input
            checked={true}
            className="c10"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <div
            checked={true}
            className="c11 c12"
          >
            <span
              className="c13 c6 lni lni-checkmark"
            />
          </div>
          <span
            className="c14 c15"
          >
            Item 1
          </span>
        </label>
      </li>
      <li
        className="c8"
      >
        <label
          className="c9"
        >
          <input
            checked={false}
            className="c10"
            onChange={[Function]}
            type="checkbox"
            value="2"
          />
          <div
            checked={false}
            className="c11 c16"
          />
          <span
            className="c14 c15"
          >
            Item 2
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`CheckboxDropdown should call default props onClick function when no onClick property is passed 1`] = `
.c6 {
  text-align: center;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c5 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        
      </span>
      <span
        className="c5 c6 lni lni-chevron-down"
      />
    </button>
  </div>
</div>
`;

exports[`CheckboxDropdown should have onClick listener 1`] = `
.c6 {
  text-align: center;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c5 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        
      </span>
      <span
        className="c5 c6 lni lni-chevron-down"
      />
    </button>
  </div>
</div>
`;

exports[`CheckboxDropdown should render 1`] = `
.c6 {
  text-align: center;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c5 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        
      </span>
      <span
        className="c5 c6 lni lni-chevron-down"
      />
    </button>
  </div>
</div>
`;

exports[`CheckboxDropdown should render in open state 1`] = `
.c7 {
  text-align: center;
}

.c11 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c13 {
  display: inline-block;
  width: 1em;
  height: 1em;
  border: 1px solid #999;
  background: transparent;
  text-align: center;
  border-radius: 4px;
  line-height: 1em;
}

.c15 {
  color: #999;
  margin-left: 1.5em;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  margin: 0.25em 0;
}

.c10:hover .c12 {
  border-color: #333;
}

.c10:hover .c14 {
  color: #666;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c8 {
  width: 100%;
  padding: 0;
  border: 1px solid;
  border-top: none;
  position: absolute;
  top: 22px;
  border-radius: 0 0 4px 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
  z-index: 1;
  box-sizing: border-box;
}

.c6 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c9 {
  padding-top: 1em;
  padding-left: 0.5em;
}

.c5 {
  font-size: 1.25em;
  font-weight: bold;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        
        <span
          className="c5"
        >
          0 selected
        </span>
      </span>
      <span
        className="c6 c7 lni lni-chevron-up"
      />
    </button>
    <ul
      className="c8"
      data-item-list-test={true}
    >
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 1
          </span>
        </label>
      </li>
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="2"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 2
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`CheckboxDropdown should render label 1`] = `
.c7 {
  text-align: center;
}

.c11 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c13 {
  display: inline-block;
  width: 1em;
  height: 1em;
  border: 1px solid #999;
  background: transparent;
  text-align: center;
  border-radius: 4px;
  line-height: 1em;
}

.c15 {
  color: #999;
  margin-left: 1.5em;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  margin: 0.25em 0;
}

.c10:hover .c12 {
  border-color: #333;
}

.c10:hover .c14 {
  color: #666;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c8 {
  width: 100%;
  padding: 0;
  border: 1px solid;
  border-top: none;
  position: absolute;
  top: 22px;
  border-radius: 0 0 4px 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
  z-index: 1;
  box-sizing: border-box;
}

.c6 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c9 {
  padding-top: 1em;
  padding-left: 0.5em;
}

.c5 {
  font-size: 1.25em;
  font-weight: bold;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        hello
        <span
          className="c5"
        >
          0 selected
        </span>
      </span>
      <span
        className="c6 c7 lni lni-chevron-up"
      />
    </button>
    <ul
      className="c8"
      data-item-list-test={true}
    >
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 1
          </span>
        </label>
      </li>
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="2"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 2
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`CheckboxDropdown should render react node label 1`] = `
.c7 {
  text-align: center;
}

.c11 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c13 {
  display: inline-block;
  width: 1em;
  height: 1em;
  border: 1px solid #999;
  background: transparent;
  text-align: center;
  border-radius: 4px;
  line-height: 1em;
}

.c15 {
  color: #999;
  margin-left: 1.5em;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  margin: 0.25em 0;
}

.c10:hover .c12 {
  border-color: #333;
}

.c10:hover .c14 {
  color: #666;
}

.c3 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c1 {
  position: relative;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid;
  border-radius: 4px;
  padding: 1.25em;
  cursor: pointer;
}

.c8 {
  width: 100%;
  padding: 0;
  border: 1px solid;
  border-top: none;
  position: absolute;
  top: 22px;
  border-radius: 0 0 4px 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
  z-index: 1;
  box-sizing: border-box;
}

.c6 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
}

.c4 {
  color: #999;
  font-size: 1.25em;
  font-weight: bold;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c9 {
  padding-top: 1em;
  padding-left: 0.5em;
}

.c5 {
  font-size: 1.25em;
  font-weight: bold;
}

<div
  className="c0"
  fontFamily="openSans"
  size="normal"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      color=""
      data-button-test={true}
      onClick={[Function]}
      type="button"
    >
      
      <span
        className="c4"
      >
        <div>
          hello
        </div>
        <span
          className="c5"
        >
          0 selected
        </span>
      </span>
      <span
        className="c6 c7 lni lni-chevron-up"
      />
    </button>
    <ul
      className="c8"
      data-item-list-test={true}
    >
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 1
          </span>
        </label>
      </li>
      <li
        className="c9"
      >
        <label
          className="c10"
        >
          <input
            checked={false}
            className="c11"
            onChange={[Function]}
            type="checkbox"
            value="2"
          />
          <div
            checked={false}
            className="c12 c13"
          />
          <span
            className="c14 c15"
          >
            Item 2
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;
